Plugins/GitUIPluginInterfaces/CredentialsManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+using AdysTech.CredentialManager;
+using JetBrains.Annotations;
+
+namespace GitUIPluginInterfaces
+{
+    public interface ICredentialsManager
+    {
+        void Save();
+    }
+
+    public class CredentialsManager : ICredentialsManager
+    {
+        private static ConcurrentDictionary<string, NetworkCredential> Credentials { get; } = new ConcurrentDictionary<string, NetworkCredential>();
+        private readonly Func<string> _getWorkingDir;
+
+        public CredentialsManager()
+        {
+        }
+
+        protected CredentialsManager(Func<string> getWorkingDir)
+        {
+            _getWorkingDir = getWorkingDir;
+        }
+
+        public void Save()
+        {
+            var credentials = Credentials.ToList();
+            if (credentials.Count < 1)
+            {
+                return;
+            }
+
+            Credentials.Clear();
+
+            foreach (var networkCredentials in credentials)
+            {
+                AdysTechCredentialManagerWrapper.UpdateCredentials(networkCredentials.Key,
+                    networkCredentials.Value.UserName,
```

Comment:
Bug what I have mentioned in my pull request is here. Value can be equal null. It is need to use Value?.UserName and Value?.Password.
Test scenario:
1. Remove all settings from Credential manager
2. Open settings
3. Select Local settings
4. Apply settings

p.s. sorry for my long absence. Unfortunately I can't check plugin because our company has enabled two-factor authentication and login/password authentication does not work now.

Reply:
Unfortunately other settings in plugin work incorrect when user switches between different setting levels. I have fixed only new credential settings and therefore Value can be empty.

Reply:
> Bug what I have mentioned in my pull request is here

Thank you for the repro steps. Fixed.

> Unfortunately other settings in plugin work incorrect when user switches between different setting levels.

That's an existing bug. I'll fix it while I am at it.